python/ray/operator.py

```diff
@@ -0,0 +1,104 @@
+"""
+Ray operator for Kubernetes.
+
+Reads ray cluster config from a k8s ConfigMap, starts a ray head node pod using
+create_or_update_cluster(), then runs an autoscaling loop in the operator pod
+executing this script. Writes autoscaling logs to the directory
+/root/ray-operator-logs.
+
+In this setup, the ray head node does not run an autoscaler. It is important
+NOT to supply an --autoscaling-config argument to head node's ray start command
+in the cluster config when using this operator.
+
+To run, first create a ConfigMap named ray-operator-configmap from a ray
+cluster config. Then apply the manifest at python/ray/autoscaler/kubernetes/operator_configs/operator_config.yaml
+
+For example:
+kubectl create namespace raytest
+kubectl -n raytest create configmap ray-operator-configmap --from-file=python/ray/autoscaler/kubernetes/operator_configs/test_cluster_config.yaml
+kubectl -n raytest apply -f python/ray/autoscaler/kubernetes/operator_configs/operator_config.yaml
+""" # noqa
+import os
+from typing import Any, Dict
+
+import kubernetes
```

Comment:
We need to include this to the package requirements. (maybe not the default one)

Reply:
yeah we can add a target like `ray[k8s]`

Reply:
> We need to include this to the package requirements. (maybe not the default one)

Err, sorry: 'this' being the `kubernetes` module?
Also -- just unfamiliar with things -- which package requirements? 
`kubernetes` is currently listed in `ray/python/requirements.txt` and `ray/docker/ray-deps/Dockerfile`

Reply:
Ah it needs to be added to `setup.py` so that people can run `pip install ray[k8s]` to get the deps. You can see the example for `ray[serve]` here:
https://github.com/ray-project/ray/blob/master/python/setup.py#L107

Reply:
Got it, thanks! Can I go ahead and add a `"k8s": ["kubernetes"]` item there?